exchanges/kucoin/kucoin_wrapper.go

```diff
@@ -0,0 +1,1383 @@
+			return err
+		}
+		for x := range ticks {
+			pair, err := currency.NewPairFromString(ticks[x].Symbol)
+			if err != nil {
+				return err
+			}
+			err = ticker.ProcessTicker(&ticker.Price{
+				Last:         ticks[x].LastTradePrice,
+				High:         ticks[x].HighPrice,
+				Low:          ticks[x].LowPrice,
+				Volume:       ticks[x].VolumeOf24h,
+				Ask:          ticks[x].IndexPrice,
+				Bid:          ticks[x].MarkPrice,
+				Pair:         pair,
+				ExchangeName: ku.Name,
+				AssetType:    assetType,
+			})
+			if err != nil {
+				return err
+			}
+		}
+		return nil
+	case asset.Spot, asset.Margin:
+		ticks, err := ku.GetTickers(ctx)
+		if err != nil {
+			return err
+		}
+		for t := range ticks.Tickers {
+			pair, err := currency.NewPairFromString(ticks.Tickers[t].Symbol)
+			if err != nil {
+				return err
+			}
+			tick := &ticker.Price{
+				Last:         ticks.Tickers[t].Last,
+				High:         ticks.Tickers[t].High,
+				Low:          ticks.Tickers[t].Low,
+				Volume:       ticks.Tickers[t].Volume,
+				Ask:          ticks.Tickers[t].Buy,
+				Bid:          ticks.Tickers[t].Sell,
+				Pair:         pair,
+				ExchangeName: ku.Name,
+				AssetType:    assetType,
+				LastUpdated:  ticks.Time.Time(),
+			}
+			err = ticker.ProcessTicker(tick)
+			if err != nil {
+				return err
+			}
+			tick.AssetType = asset.Margin
+			err = ticker.ProcessTicker(tick)
+			if err != nil {
+				return err
+			}
+		}
+	default:
+		return fmt.Errorf("%w %v", asset.ErrNotSupported, assetType)
+	}
+	return nil
+}
+
+// FetchTicker returns the ticker for a currency pair
+func (ku *Kucoin) FetchTicker(ctx context.Context, p currency.Pair, assetType asset.Item) (*ticker.Price, error) {
+	p, err := ku.FormatExchangeCurrency(p, assetType)
+	if err != nil {
+		return nil, err
+	}
+	tickerNew, err := ticker.GetTicker(ku.Name, p, assetType)
+	if err != nil {
+		return ku.UpdateTicker(ctx, p, assetType)
+	}
+	return tickerNew, nil
+}
+
+// FetchOrderbook returns orderbook base on the currency pair
+func (ku *Kucoin) FetchOrderbook(ctx context.Context, pair currency.Pair, assetType asset.Item) (*orderbook.Base, error) {
+	pair, err := ku.FormatExchangeCurrency(pair, assetType)
+	if err != nil {
+		return nil, err
+	}
+	ob, err := orderbook.Get(ku.Name, pair, assetType)
+	if err != nil {
+		return ku.UpdateOrderbook(ctx, pair, assetType)
+	}
+	return ob, nil
+}
+
+// UpdateOrderbook updates and returns the orderbook for a currency pair
+func (ku *Kucoin) UpdateOrderbook(ctx context.Context, pair currency.Pair, assetType asset.Item) (*orderbook.Base, error) {
+	err := ku.CurrencyPairs.IsAssetEnabled(assetType)
+	if err != nil {
+		return nil, err
+	}
+	pair, err = ku.FormatExchangeCurrency(pair, assetType)
+	if err != nil {
+		return nil, err
+	}
+	var ordBook *Orderbook
+	switch assetType {
+	case asset.Futures:
+		ordBook, err = ku.GetFuturesOrderbook(ctx, pair.String())
+	case asset.Spot, asset.Margin:
+		if ku.IsRESTAuthenticationSupported() {
```

Comment:
This should be inverted.

This is something that would become clear if you ran the application. I keep saying this.
```
[ERROR] | ORDERBOOK | 02/03/2023 14:44:45 | Failed to get REST orderbook. Error: Kucoin REST or Websocket authentication support is not enabled
[ERROR] | ORDERBOOK | 02/03/2023 14:44:45 | Failed to get REST orderbook. Error: Kucoin REST or Websocket authentication support is not enabled
[ERROR] | ORDERBOOK | 02/03/2023 14:44:45 | Failed to get REST orderbook. Error: Kucoin REST or Websocket authentication support is not enabled
[ERROR] | ORDERBOOK | 02/03/2023 14:44:45 | Failed to get REST orderbook. Error: Kucoin REST or Websocket authentication support is not enabled
[ERROR] | ORDERBOOK | 02/03/2023 14:44:45 | Failed to get REST orderbook. Error: Kucoin REST or Websocket authentication support is not enabled
[ERROR] | ORDERBOOK | 02/03/2023 14:44:45 | Failed to get REST orderbook. Error: Kucoin REST or Websocket authentication support is not enabled
[ERROR] | ORDERBOOK | 02/03/2023 14:44:45 | Failed to get REST orderbook. Error: Kucoin REST or Websocket authentication support is not enabled
[ERROR] | ORDERBOOK | 02/03/2023 14:44:45 | Failed to get REST orderbook. Error: Kucoin REST or Websocket authentication support is not enabled
[ERROR] | ORDERBOOK | 02/03/2023 14:44:45 | Failed to get REST orderbook. Error: Kucoin REST or Websocket authentication support is not enabled
[ERROR] | ORDERBOOK | 02/03/2023 14:44:45 | Failed to get REST orderbook. Error: Kucoin REST or Websocket authentication support is not enabled
[ERROR] | ORDERBOOK | 02/03/2023 14:44:45 | Failed to get REST orderbook. Error: Kucoin REST or Websocket authentication support is not enabled
```